Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm

```diff
@@ -82,17 +140,40 @@
 #endif
     }
 }
+#endif
 
 void ProcessLauncher::launchProcess()
 {
     ASSERT(!m_xpcConnection);
 
-    const char* name = serviceName(m_launchOptions, m_client);
+#if USE(EXTENSIONKIT)
+    auto handler = [](ThreadSafeWeakPtr<ProcessLauncher> weakProcessLauncher, _SEExtensionProcess* process, const char* name, NSError* error)
+    {
+        if (!weakProcessLauncher.get())
+            return;
+        if (error)
+            NSLog(@"Error launching process %@ error %@", process, error);
+        callOnMainRunLoop([weakProcessLauncher = weakProcessLauncher, name = name, process = RetainPtr<_SEExtensionProcess>(process)] {
```

Comment:
The use of `callOnMainRunLoop` makes it seem like some of this code might run off the main thread. Do we need to be doing an isolated copy of name here? Or adding asserts that the other code referencing this `name` (which comes from `serviceNameAndIdentifier`) is on the right thread?